lib/ash/generator/generator.ex

```diff
@@ -505,12 +505,27 @@ defmodule Ash.Generator do
             fn _changeset, record ->
               {:ok, after_action.(record)}
             end
+          else
+            # Do nothing
+            fn _changeset, record -> {:ok, record} end
```

Comment:
I'll merge but I'm going to figure out how to make this not necessary, as adding hooks that aren't needed can prevent internal optimizations etc.